src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1702,6 +1702,15 @@ multi_join_restriction_hook(PlannerInfo *root,
 							JoinType jointype,
 							JoinPathExtraData *extra)
 {
+	if (bms_is_empty(innerrel->relids) || bms_is_empty(outerrel->relids))
+	{
+		/*
```

Comment:
We do not expect empty relids and we cannot process them as well. Still,  ignoring such JoinRestriction is preferable for two reasons. First, this might be a query that doesn't rely on JoinRestrictions at all (e.g., local query). Second, even if we add these restrictions, we cannot process them (and likely to segfault if we allow as-is).

Reply:
(This is to include @onurctirtir's feedback)